src/DynamoDb/SetValue.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace Aws\DynamoDb;
+
+use GuzzleHttp\ToArrayInterface;
+
+/**
+ * Special object to represent a DynamoDB set (SS/NS/BS) value.
+ */
+class SetValue implements \JsonSerializable, \ArrayAccess, \Countable, \IteratorAggregate, ToArrayInterface
```

Comment:
Looking at how Set works now, I wonder if we actually should make it immutable. This would push validation into the constructor of the Set, cast each value to string (which it looks like is a requirement), and we could drop getValues(). Also note that the other wrappers are immutable.